## imports and tables

In [17]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import scipy.stats as stats

In [4]:
heart = pd.read_csv('../data/heart_failure_clinical_records_dataset.csv')

In [5]:
heart.head()

age  anaemia  creatinine_phosphokinase  diabetes  ejection_fraction  \
0  75.0        0                       582         0                 20   
1  55.0        0                      7861         0                 38   
2  65.0        0                       146         0                 20   
3  50.0        1                       111         0                 20   
4  65.0        1                       160         1                 20   

   high_blood_pressure  platelets  serum_creatinine  serum_sodium  sex  \
0                    1  265000.00               1.9           130    1   
1                    0  263358.03               1.1           136    1   
2                    0  162000.00               1.3           129    1   
3                    0  210000.00               1.9           137    1   
4                    0  327000.00               2.7           116    0   

   smoking  time  DEATH_EVENT  
0        0     4            1  
1        0     6            1  
2        1     7            1  
3        0     7            1  
4        0     8            1

In [6]:
heart.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 299 entries, 0 to 298
Data columns (total 13 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   age                       299 non-null    float64
 1   anaemia                   299 non-null    int64  
 2   creatinine_phosphokinase  299 non-null    int64  
 3   diabetes                  299 non-null    int64  
 4   ejection_fraction         299 non-null    int64  
 5   high_blood_pressure       299 non-null    int64  
 6   platelets                 299 non-null    float64
 7   serum_creatinine          299 non-null    float64
 8   serum_sodium              299 non-null    int64  
 9   sex                       299 non-null    int64  
 10  smoking                   299 non-null    int64  
 11  time                      299 non-null    int64  
 12  DEATH_EVENT               299 non-null    int64  
dtypes: float64(3), int64(10)
memory usage: 30.5 KB


In [20]:
anemia = heart[heart['anaemia'] == 1]
anemia.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 129 entries, 3 to 293
Data columns (total 13 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   age                       129 non-null    float64
 1   anaemia                   129 non-null    int64  
 2   creatinine_phosphokinase  129 non-null    int64  
 3   diabetes                  129 non-null    int64  
 4   ejection_fraction         129 non-null    int64  
 5   high_blood_pressure       129 non-null    int64  
 6   platelets                 129 non-null    float64
 7   serum_creatinine          129 non-null    float64
 8   serum_sodium              129 non-null    int64  
 9   sex                       129 non-null    int64  
 10  smoking                   129 non-null    int64  
 11  time                      129 non-null    int64  
 12  DEATH_EVENT               129 non-null    int64  
dtypes: float64(3), int64(10)
memory usage: 14.1 KB


In [21]:
no_anemia = heart[heart['anaemia'] == 0]
no_anemia.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 170 entries, 0 to 298
Data columns (total 13 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   age                       170 non-null    float64
 1   anaemia                   170 non-null    int64  
 2   creatinine_phosphokinase  170 non-null    int64  
 3   diabetes                  170 non-null    int64  
 4   ejection_fraction         170 non-null    int64  
 5   high_blood_pressure       170 non-null    int64  
 6   platelets                 170 non-null    float64
 7   serum_creatinine          170 non-null    float64
 8   serum_sodium              170 non-null    int64  
 9   sex                       170 non-null    int64  
 10  smoking                   170 non-null    int64  
 11  time                      170 non-null    int64  
 12  DEATH_EVENT               170 non-null    int64  
dtypes: float64(3), int64(10)
memory usage: 18.6 KB


In [11]:
no_anemia_men = no_anemia[no_anemia['sex'] == 1]
no_anemia_women = no_anemia[no_anemia['sex'] == 0]

In [12]:
anemia_men = anemia[anemia['sex'] == 1]
anemia_women = anemia[anemia['sex'] == 0]

In [22]:
no_anemia_men.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 117 entries, 0 to 298
Data columns (total 13 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   age                       117 non-null    float64
 1   anaemia                   117 non-null    int64  
 2   creatinine_phosphokinase  117 non-null    int64  
 3   diabetes                  117 non-null    int64  
 4   ejection_fraction         117 non-null    int64  
 5   high_blood_pressure       117 non-null    int64  
 6   platelets                 117 non-null    float64
 7   serum_creatinine          117 non-null    float64
 8   serum_sodium              117 non-null    int64  
 9   sex                       117 non-null    int64  
 10  smoking                   117 non-null    int64  
 11  time                      117 non-null    int64  
 12  DEATH_EVENT               117 non-null    int64  
dtypes: float64(3), int64(10)
memory usage: 12.8 KB


In [23]:
no_anemia_women.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 53 entries, 8 to 296
Data columns (total 13 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   age                       53 non-null     float64
 1   anaemia                   53 non-null     int64  
 2   creatinine_phosphokinase  53 non-null     int64  
 3   diabetes                  53 non-null     int64  
 4   ejection_fraction         53 non-null     int64  
 5   high_blood_pressure       53 non-null     int64  
 6   platelets                 53 non-null     float64
 7   serum_creatinine          53 non-null     float64
 8   serum_sodium              53 non-null     int64  
 9   sex                       53 non-null     int64  
 10  smoking                   53 non-null     int64  
 11  time                      53 non-null     int64  
 12  DEATH_EVENT               53 non-null     int64  
dtypes: float64(3), int64(10)
memory usage: 5.8 KB


In [24]:
anemia_men.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 77 entries, 3 to 293
Data columns (total 13 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   age                       77 non-null     float64
 1   anaemia                   77 non-null     int64  
 2   creatinine_phosphokinase  77 non-null     int64  
 3   diabetes                  77 non-null     int64  
 4   ejection_fraction         77 non-null     int64  
 5   high_blood_pressure       77 non-null     int64  
 6   platelets                 77 non-null     float64
 7   serum_creatinine          77 non-null     float64
 8   serum_sodium              77 non-null     int64  
 9   sex                       77 non-null     int64  
 10  smoking                   77 non-null     int64  
 11  time                      77 non-null     int64  
 12  DEATH_EVENT               77 non-null     int64  
dtypes: float64(3), int64(10)
memory usage: 8.4 KB


In [25]:
anemia_women.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 52 entries, 4 to 289
Data columns (total 13 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   age                       52 non-null     float64
 1   anaemia                   52 non-null     int64  
 2   creatinine_phosphokinase  52 non-null     int64  
 3   diabetes                  52 non-null     int64  
 4   ejection_fraction         52 non-null     int64  
 5   high_blood_pressure       52 non-null     int64  
 6   platelets                 52 non-null     float64
 7   serum_creatinine          52 non-null     float64
 8   serum_sodium              52 non-null     int64  
 9   sex                       52 non-null     int64  
 10  smoking                   52 non-null     int64  
 11  time                      52 non-null     int64  
 12  DEATH_EVENT               52 non-null     int64  
dtypes: float64(3), int64(10)
memory usage: 5.7 KB


## Hypothesis testing

### Do people with anemia and poeple without anemia have differing platelet counts?

#### H0: Platelet counts in people who have anemia and people who don't ARE NOT statistically different.

#### H1: Platelet counts in people who have anemia and people who don't ARE statistically significant

#### P-VALUE: 0.05

In [18]:
stats.ttest_ind(anemia['platelets'], no_anemia['platelets'])

Ttest_indResult(statistic=-0.7553108790369495, pvalue=0.450661035974825)

####  With a p-value of 0.45, I do not have enough evidence to reject the null hypothesis